Src/VimCore/RegisterMap.fs

```diff
@@ -100,9 +100,19 @@ type internal RegisterMap (_map : Map<RegisterName, Register>) =
     /// Updates the given register with the specified value.  This will also update 
     /// other registers based on the type of update that is being performed.  See 
     /// :help registers for the full details
-    member x.SetRegisterValue (reg : Register) regOperation (value : RegisterValue) = 
+    member x.SetRegisterValue (register : Option<Register>) regOperation (value : RegisterValue) (clipboardOptions : Vim.ClipboardOptions) = 

```

Comment:
I feel like this might be the wrong approach.  The `RegisterMap` and `Register` types are designed to be unaware of settings.  All of the settings aware code was meant to be layered on top of them.  The approach I had in mind for this fix was to build on top of the existing code which attempts to access `IRegisterMap` in a settings aware manner.